torch/nn/modules/activation.py

```diff
@@ -886,6 +886,24 @@ def extra_repr(self) -> str:
         return str(self.lambd)
 
 
+def _arg_cuda_or_cpu(x: Optional[torch.Tensor]) -> bool:
+    if x is None:
+        return True
+    else:
+        return x.is_cuda or 'cpu' in str(x.device)
+
+    return False
```

Comment:
```suggestion
def _arg_cuda_or_cpu(x: Optional[torch.Tensor]) -> bool:
    if x is None:
        return True
    return x.is_cuda or 'cpu' in str(x.device)
```

Reply:
Putting the conditional into the else path is required to get type refinement for static torchscript typing. 

Type refinement is only active for the duration of the if/else from what I understand, and then reverts to pre-if typing, i.e., it will complain that a NoneType (which is part of Optional[tensor]) does not have is_cuda as attribute.